test/e2e/framework/helpers/k8s/k8s_utils.go

```diff
@@ -1779,3 +1822,96 @@ func (k *KubeCtl) DeleteStorageClass(scName string) error {
+}
+
+func (k *KubeCtl) GetSecretValue(namespace, secretName, key string) (string, error) {
+	var secret *v1.Secret
+	var err error
+
+	// Retry loop in case secret is not yet populated
+	for i := 0; i < 5; i++ {
+		secret, err = k.clientSet.CoreV1().Secrets(namespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+		if err != nil {
+			return "", err
+		}
+
+		// Check if the secret contains data
+		if len(secret.Data) > 0 {
+			value, ok := secret.Data[key]
+			if !ok {
+				return "", fmt.Errorf("key %s not found in secret %s", key, secretName)
+			}
+			return string(value), nil
+		}
+
+		// Wait before retrying
+		time.Sleep(2 * time.Second)
+	}
+
+	return "", fmt.Errorf("secret %s has no data after retries", secretName)
+}
+
+func (k *KubeCtl) GetCurrentContext() (string, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+
+	rawConfig, err := kubeConfig.RawConfig()
+	if err != nil {
+		return "", err
+	}
+	return rawConfig.CurrentContext, nil
+}
+
+func (k *KubeCtl) GetCurrentCluster() (string, error) {
+	// get the current context cluster name
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+	rawConfig, err := kubeConfig.RawConfig()
+	if err != nil {
+		return "", err
+	}
+	return rawConfig.Contexts[rawConfig.CurrentContext].Cluster, nil
+}
+
+func (k *KubeCtl) GetClusterCA() ([]byte, error) {
+	// Get the current clustername
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+	rawConfig, err := kubeConfig.RawConfig()
+	if err != nil {
+		return nil, err
+	}
+	return rawConfig.Clusters[rawConfig.Contexts[rawConfig.CurrentContext].Cluster].CertificateAuthorityData, nil
+}
+
+func (k *KubeCtl) GetClusterServer() (string, error) {
+	// Get the current clustername
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+	rawConfig, err := kubeConfig.RawConfig()
+	if err != nil {
+		return "", err
+	}
+	return rawConfig.Clusters[rawConfig.Contexts[rawConfig.CurrentContext].Cluster].Server, nil
+}
+
+func (k *KubeCtl) UpdateConfig(newConf *rest.Config) error {
+	_, err := clientcmd.BuildConfigFromFlags("", clientcmd.RecommendedHomeFile)
+	if err != nil {
+		return err
+	}
+	newClientset, err := kubernetes.NewForConfig(newConf)
+	if err != nil {
+		return err
+	}
+	k.clientSet = newClientset
+	k.kubeConfig = newConf
+	return nil
+}
```

Comment:
All these methods seem to be unused. Am I missing something?

Reply:
Removed